src/Avalonia.Controls/ItemsSourceView.cs

```diff
@@ -42,27 +42,22 @@ public class ItemsSourceView : INotifyCollectionChanged, IDisposable
         public ItemsSourceView(IEnumerable source)
         {
             source = source ?? throw new ArgumentNullException(nameof(source));
-
-            if (source is IList list)
-            {
-                _inner = list;
-            }
-            else if (source is IEnumerable<object> objectEnumerable)
-            {
-                _inner = new List<object>(objectEnumerable);
-            }
-            else
+            _inner = source switch
             {
-                _inner = new List<object>(source.Cast<object>());
-            }
-
-            ListenToCollectionChanges();
+                ItemsSourceView => throw new ArgumentException("Cannot wrap an existing ItemsSourceView.", nameof(source)),
```

Comment:
This feature is not supported on our SDK, please update

Reply:
![image](https://user-images.githubusercontent.com/53405089/131228369-d2d68f62-10c6-4f31-85b2-9a51bb1a3249.png)

Reply:
Balls. We really need to update our SDK.

Reply:
Fixed.